scenario/state.py

```diff
@@ -749,6 +960,16 @@ class Port:
     port: Optional[int] = None
     """The port to open. Required for TCP and UDP; not allowed for ICMP."""
 
+    # NOTE: This can be replaced with dataclass(kw_only=True) in Python 3.10+
+    def __init__(
+        self,
+        *,
+        protocol: _RawPortProtocolLiteral,
+        port: Optional[int] = None,
```

Comment:
The issue I have with `Port` is that I don't see how people remember if it's `Port("tcp", 22)` or `Port(22, "tcp")`. It's 'obvious' to you and me because we've worked with them before, but I feel that there's not really any inherent logic in having it be protocol-port rather than port-protocol. I think that's especially the case since the name is "port" and people often say "port" meaning "port number" so the first thing that comes to mind is the number, not the protocol. On the other side, since `icmp` doesn't have a number, it can't be `Port(port, protocol`), but I don't think people are thinking about that.

Having `protocol` default to `tcp` seems reasonable, except that then `Port()` is ok because `port` has a default value, and it would have to raise an exception, and I imagine that's not going to be detectable by linters/static checks/IDEs.

Reply:
Yeah, that's fair re the order not being obvious. I still think `Port(port=42)` is quite useful and we should do that bit, even if it means a runtime error rather than a linter error in the `Port()` case.

Reply:
Perhaps we could consider making _Port private and adding TCPPort, UDPPort, ICMPPort? Would resolve the signature awkwardness